### PR TITLE
Report k0s binary size in job summary

### PR DESCRIPTION
**Pull request by ncopa**
fixes https://github.com/k0sproject/k0s/issues/3191

Also add a fix for detecting executable suffix while at it.

## Description

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [ ] Manual test
- [ ] Auto test added

## Checklist:

- [ ] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [ ] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by ncopa** [comment]
I had a look at this, and it seems a bit complicated to give permissions for a forked repo to post comments in the PR. Basically, we'd need to set up a proxy service. I don't think it is worth it.

As an alternate option, we can post the binary size as a [GITHUB_STEP_SUMMARY](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary). It is stupid and simple, even if not as convenient as having a PR comment.

File: .github/workflows/build-k0s.yml [hunk 1]
```diff
@@ -71,14 +71,14 @@ jobs:
           make bindata
           make --touch codegen
           make build
+          echo "k0s${{ steps.build-prepare.outputs.executable-suffix }} binary size: **$(du -sh k0s${{ steps.build-prepare.outputs.executable-suffix }} | cut -f1)**" >>$GITHUB_STEP_SUMMARY
```
Comment (mikhail-sakhnov) [inline .github/workflows/build-k0s.yml:74]: can we somehow post it as a comment to the PR?

Comment (ncopa) [reply .github/workflows/build-k0s.yml:74]: We can, but it require us to run a proxy service in our infra.
https://github.com/marketplace/actions/add-pr-comment#proxy-for-fork-based-prs

**Comment by ncopa** [comment]
Closing this, since this is not the way we will solve it

**Pull request closed without merging**